Sources/SKCore/CompilationDatabaseBuildSystem.swift

```diff
@@ -59,22 +59,17 @@ extension CompilationDatabaseBuildSystem: BuildSystem {
     indexStorePath?.parentDirectory.appending(component: "IndexDatabase")
   }
 
-  public func settings(for uri: DocumentURI, _ language: Language) -> FileBuildSettings? {
-    guard let url = uri.fileURL else {
-      // We can't determine build settings for non-file URIs.
-      return nil
-    }
-    guard let db = database(for: url),
-          let cmd = db[url].first else { return nil }
-    return FileBuildSettings(
-      compilerArguments: Array(cmd.commandLine.dropFirst()),
-      workingDirectory: cmd.directory,
-      language: language)
+  /// Register the given file for build-system level change notifications, such
+  /// as command line flag changes, dependency changes, etc.
+  ///
+  /// IMPORTANT: When first receiving a register request, the `BuildSystem`
+  /// MUST eventually inform its delegate of any initial settings for the given file
+  /// via the `fileBuildSettingsChanged` method.
+  public func registerForChangeNotifications(for uri: DocumentURI, language: Language) {
+    let settings = self._settings(for: uri, language)
+    self.delegate?.fileBuildSettingsChanged([uri: FileBuildSettingsChange(settings)])
```

Comment:
Should be done asynchronously - e.g. on a global dispatch queue.  Clients usually don't expect to be called synchronously like this.

Reply:
Hmm at least at the moment this is ok for the LSP itself since the BuildSystemManager handles it just fine via its queues, I guess it's still worth doing this for tests?

Reply:
Yes, and for future proofing.

Reply:
Done